controller_interface/include/controller_interface/controller_interface.hpp

```diff
@@ -89,6 +89,11 @@ class ControllerInterface : public rclcpp_lifecycle::node_interfaces::LifecycleN
   return_type
   init(const std::string & controller_name);
 
+  CONTROLLER_INTERFACE_PUBLIC
+  virtual
+  return_type
+  init(const std::string & controller_name, rclcpp::NodeOptions & node_options);
```

Comment:
can you please try removing the `controller_name` argument? 

We were planning on dropping it as this name is set at loading time and available from the `LifecycleNodeInterface` part of `ControllerInterface`, this argument is currently unused in all controllers.

Ref: https://github.com/ros-controls/ros2_control/issues/251

Reply:
Do you mean in both versions of the `init` function ?

Reply:
It seems to me that the controller_name is set at load time indeed, but by using the `controller_name` argument in the `ControllerManager::add_controller_impl` function (see [line 535](https://github.com/ros-controls/ros2_control/blob/f33d95d840ca8403dd5ea2f4e499826778ecc977/controller_manager/src/controller_manager.cpp#L535)).

And as far as I know, the `ControllerInterface` is a child of `LifeCycleNodeInterface`, just like `LifeCycleNode`, but doesn't have the member variables of the latter.

So I don't see how I could initialize this argument. Did I miss something ?

Reply:
you are right, this option was only valid when we were inheriting from LifecycleNode, not the interface, the argument is actually being used